test/data/fle2-find/encrypted-payload.json

```diff
@@ -0,0 +1,15 @@
+{
+   "documents": [
+      {
+         "_id": 1,
+         "ssn": {
+            "$binary": {
+               "base64": "BYkAAAAFZAAgAAAAAHb62aV7+mqmaGcotPLdG3KP7S8diFwWMLM/5rYtqLrEBXMAIAAAAAAVJ6OWHRv3OtCozHpt3ZzfBhaxZirLv3B+G8PuaaO4EgVjACAAAAAAsZXWOWA+UiCBbrJNB6bHflB/cn7pWSvwWN2jw4FPeIUSY20AAAAAAAAAAAAA",
```

Comment:
I expect this to equal the base64 in `docWithTransformedPlaceholders` from the first kFind case in the [captured payloads](https://gist.github.com/kevinAlbs/cba611fe0d120b3f67c6bee3195d4ce6):

```
roundTripTest ... begin
... operation = Operation::kFind
... algorithm = FLE2AlgorithmInt::kEquality
... inputDoc = {"plainText":"sample","encrypted":"value123"}
... docWithPlaceholder (payload 3) = {"plainText":"sample","encrypted":{"$binary":{"base64":"A2EAAAAQdAACAAAAEGEAAgAAAAVraQAQAAAABBI0VngSNJh2EjQSNFZ4kBIFa3UAEAAAAASrze+rEjSYdhI0EjRWeJASAnYACQAAAHZhbHVlMTIzABJjbQAAAAAAAAAAAAA=","subType":"6"}}}
... docWithTransformedPlaceholders (payload 4 or 5 or 6) = {"plainText":"sample","encrypted":{"$binary":{"base64":"BYkAAAAFZAAgAAAAAPtVteJQAlgb2YMa/+7YWH00sbQPyt7L6Rb8OwBdMmL2BXMAIAAAAAAd44hgVKnEnTFlwNVC14oyc9OZOTspeymusqkRQj57nAVjACAAAAAA19X9v9NlWidu/wR5/C/7WUV54DfL5CkNmT5WYrhxdDcSY20AAAAAAAAAAAAA","subType":"6"}}}
... finalDoc (payload 7 or 6) = {"plainText":"sample","encrypted":{"$binary":{"base64":"BYkAAAAFZAAgAAAAAPtVteJQAlgb2YMa/+7YWH00sbQPyt7L6Rb8OwBdMmL2BXMAIAAAAAAd44hgVKnEnTFlwNVC14oyc9OZOTspeymusqkRQj57nAVjACAAAAAA19X9v9NlWidu/wR5/C/7WUV54DfL5CkNmT5WYrhxdDcSY20AAAAAAAAAAAAA","subType":"6"}},"__safeContent__":[]}
roundTripTest ... end
```

Why does this differ?

Reply:
1. I was using a different find command (ssn: "123-45-6789") and didn't realize we have expected outputs for everything in here.  I've updated the test to use one of these (the one using (value: 123456, specifically, since there's not a clean JSON way of expressing a regex pattern).

2. Looks like there was a legit bug because I misread edcDerivedDataToken as being the same as edcDerivedDataTokenAndCounter.  Fixed in latest commit, matches the expected output now.